runtime/src/iree-amd-aie/aie_runtime/iree_aie_router.cc

```diff
@@ -0,0 +1,319 @@
+    routingSolution.clear();
+    for (auto &ch : edges) ch.usedCapacity = 0;
+
+    // for each flow, find the shortest path from source to destination
+    // update used_capacity for the path between them
+    for (const auto &[src, dsts] : flows) {
+      // Use dijkstra to find path given current demand from the start
+      // switchbox; find the shortest paths to each other switchbox. Output is
+      // in the predecessor map, which must then be processed to get individual
+      // switchbox settings
+      assert(src.sb && "nonexistent flow source");
+      std::set<SwitchboxNode *> processed;
+      std::map<SwitchboxNode *, SwitchboxNode *> preds =
+          dijkstraShortestPaths(graph, src.sb);
+
+      // trace the path of the flow backwards via predecessors
+      // increment used_capacity for the associated channels
+      SwitchSettings switchSettings;
+      // set the input bundle for the source endpoint
+      switchSettings[*src.sb].src = src.port;
+      processed.insert(src.sb);
+      for (const PathEndPointNode &endPoint : dsts) {
+        SwitchboxNode *curr = endPoint.sb;
+        assert(curr && "endpoint has no source switchbox");
+        // set the output bundle for this destination endpoint
+        switchSettings[*curr].dsts.insert(endPoint.port);
+
+        // trace backwards until a vertex already processed is reached
+        while (!processed.count(curr)) {
+          // find the edge from the pred to curr by searching incident edges
+          SmallVector<ChannelEdge *, 10> channels;
+          graph.findIncomingEdgesToNode(*curr, channels);
+          auto *matchingCh = std::find_if(
+              channels.begin(), channels.end(),
+              [&](ChannelEdge *ch) { return ch->src == *preds[curr]; });
+          assert(matchingCh != channels.end() && "couldn't find ch");
+          // incoming edge
+          ChannelEdge *ch = *matchingCh;
+
+          // don't use fixed channels
+          while (ch->fixedCapacity.count(ch->usedCapacity)) ch->usedCapacity++;
+
+          // add the entrance port for this Switchbox
+          switchSettings[*curr].src = {getConnectingBundle(ch->bundle),
+                                       ch->usedCapacity};
+          // add the current Switchbox to the map of the predecessor
+          switchSettings[*preds[curr]].dsts.insert(
+              {ch->bundle, ch->usedCapacity});
+
+          ch->usedCapacity++;
+          // if at capacity, bump demand to discourage using this Channel
+          if (ch->usedCapacity >= ch->maxCapacity) ch->demand *= DEMAND_COEFF;
+
+          processed.insert(curr);
+          curr = preds[curr];
+        }
+      }
+      // add this flow to the proposed solution
+      routingSolution[src] = switchSettings;
+    }
+  } while (!isLegal());  // continue iterations until a legal routing is found
+
+  return routingSolution;
+}
+
+std::optional<std::map<PathEndPoint, SwitchSettings>> findPaths(
+    int maxIterations);
```

Comment:
What does this do?

Reply:
The whole algo is basically 

1. try to route using dijkstra
2. succes? cool
3. fail and less than max iterations ? dump congestion coefficients goto 1 else bail and report error

Reply:
I meant, isn't this the same function as above?

Reply:
Lolwut I have no idea how that happened. Will delete.